### ISSUE TITLE
reportGeneralTypeIssues when using type hints of aenum.Flag

### ISSUE THREAD

**Issue by mw-root** [microsoft/pyright#6201]
## Description
Pyright seems to be evaluating aenum.Flag objects as their values and not the actual class type.

The errors thrown appear to be for the object's `value` property and not the object itself.

## Code Example
```python
import aenum

@aenum.unique
class MyClass(aenum.Flag):
    FIRST = pow(2, 0)
    SECOND = pow(2, 1)

def my_func(my_arg: list[MyClass]):
    return None

my_func([MyClass.FIRST])
```

## Results
```
❯ pyright --version
pyright 1.1.332

❯ pyright
file.py
  file.py:15:10 - error: Argument of type "list[int]" cannot be assigned to parameter "my_arg" of type "list[MyClass]" in function "my_func"
    "Literal[1]" is incompatible with "MyClass" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations
```

**Comment by erictraut** [comment]
The `aenum` library is an untyped library. Do you have any type stubs installed for it, or are you relying entirely on pyright's type inference for its types?

When I install the latest version of `aenum` and run pyright on your code sample, I receive two errors. The first error (which is reported for the `MyClass` class statement) is the key to the second error:

```
Argument to class must be a base class
```

Are you not seeing this error? Perhaps you omitted it in your bug report or your actual code has a `# type: ignore` to suppress the error? This error is because the `aenum.Flag` symbol is a dynamically-generated class, and static type checkers cannot handle this sort of dynamic behavior. 

I've never heard of `aenum` before, but it appears to be similar to the stdlib `enum` module. The `Enum` and `EnumMeta` classes require significant special-cased logic in type checkers, and this special-cased logic doesn't apply to third-party modules like `aenum`, so it's possible that this library simply won't work with static type checking.

I'll note that mypy emits the same two errors for this code sample.

**Comment by erictraut** [comment]
I don't think this is a bug in pyright, so I'm going to close the issue. Feel free to respond to my questions above, and I'll help if I can.
